### ISSUE TITLE
Compiler errros on FreeBSD

### ISSUE THREAD

**Issue by driesmp** [st3w/neo#8]
When trying to build neo in a clean environment (poudriere).
I get the errors in the log attached, could you take a look?

[neo-0.6.log](https://github.com/st3w/neo/files/7726194/neo-0.6.log)

**Comment by st3w** [comment]
Thanks for being the FreeBSD guinea pig for this one!

Would you be willing to build it from source after I push a potential fix? I ask because it might take me a while to spin up a FreeBSD VM to verify any fixes from my end.

Also, what is the compiler name and version? It's actually catching things I didn't see with g++ and clang++.

**Comment by driesmp** [comment]
Sure, if you create a pull request I can pull in patches to test before you commit to main :) (I'm using the release tarball).

**Comment by driesmp** [comment]
FreeBSD's native compiler is CLANG. But I think our flags and optimizations are very strict.

**Comment by st3w** [comment]
I had a bug where I assigned the output of strtoul() to an unsigned int (strtoul return an unsigned **long**). Somehow "-Wall -Wextra -Wpedantic -Werror" was not enough to catch that on my setup, but FreeBSD+Clang caught it!

The thing that was actually causing the FreeBSD compilation to fail had to do with how some of the RNG fields were constructed.

I just pushed a fix to the main branch. Hopefully, this will work now. I'm fairly confident the fixes are good and that Clang was flagging legitimate issues.

I g2g sleep. It' like 2:30am over here. Lmk if it's working now. I'll check back tomorrow.
